src/components/radio/__tests__/RadioGroupBox.Test.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { useState } from 'react';
+import userEvent from '@testing-library/user-event';
+
+import { render, screen } from 'jest-utils';
+
+import RadioGroupBox from '../RadioGroupBox';
+
+const baseOptions = [
+  { label: 'Option A', value: 'a' },
+  { label: 'Option B', value: 'b' },
+];
+
+const followUps = {
+  a: {
+    id: 'followupA',
+    name: 'followupA',
+    options: [{ label: 'FA1', value: 'fa1' }],
+    groupLabel: 'Follow-up A',
+  },
+};
+
+const Wrapper = () => {
+  const [values, setValues] = useState<Record<string, string>>({
+    main: 'a',
+    followupA: 'fa1',
+  });
+  return (
+    <RadioGroupBox
+      id="main"
+      name="mainGroup"
+      options={baseOptions}
+      groupLabel="Main Group"
+      followUps={followUps}
+      values={values}
+      onChange={(newValues) => setValues(newValues)}
+    />
+  );
+};
+
+describe('<RadioGroupBox />', () => {
```

Comment:
What data should be sent when I:
- select `a`
- select `fa1`
- go back and select `b`
- submit the form?

Do we want to test that case?